proxy/http2/Http2ConnectionState.cc

```diff
@@ -1120,6 +1120,9 @@ Http2ConnectionState::release_stream(Http2Stream *stream)
     // We were shutting down, go ahead and terminate the session
     ua_session->destroy();
     ua_session = nullptr;
+  } else if (total_client_streams_count == 0 && http2_drain && ua_session && stream) {
+    send_goaway_frame(stream->get_id(), Http2ErrorCode::HTTP2_ERROR_NO_ERROR);
```

Comment:
There should be 2 GOWAY frames sent during graceful shutdown.  @zizhong it seems you are only doing the 2nd one. 
1) stream id - 2^31 - 1 - This signals the client that we are shutting down so that it should stop sending us more streams
2) After allowing time for any in-flight stream creation (at least one round-trip time), the server can send another GOAWAY frame with an updated last stream identifier. This ensures that a connection can be cleanly shut down without losing requests.